### PR TITLE
Fix substring simplifier

### PR DESCRIPTION
**Pull request by edmundnoble**
#qz-3454 Done

### CODE REVIEW

File: frontend/src/test/scala/quasar/std/StdLibSpec.scala [hunk 1]
```diff
@@ -167,10 +167,21 @@ abstract class StdLibSpec extends Qspec {
...
+          // restrict the range to something that will actually exercise the behavior
+          ternary(Substring(_, _, _).embed, Data.Str(str), Data.Int(-1), Data.Int(str.length), Data.Str(""))
+        }
+
         "any string and offsets" >> prop { (str: String, start0: Int, length0: Int) =>
           // restrict the range to something that will actually exercise the behavior
-          val start = start0 % 1000
-          val length = length0 % 1000
+          val start = (start0 % 1000) - 500
+          val length = (length0 % 1000) - 500
```
Comment (alissapajer) [inline frontend/src/test/scala/quasar/std/StdLibSpec.scala]: Why the ` - 500`?

Comment (alissapajer) [reply frontend/src/test/scala/quasar/std/StdLibSpec.scala]: Oh I figured it out, to get negative numbers :)

Comment (alissapajer) [reply frontend/src/test/scala/quasar/std/StdLibSpec.scala]: Never mind, `%` returns negative numbers. Asking again. :)

Comment (edmundnoble) [reply frontend/src/test/scala/quasar/std/StdLibSpec.scala]: Heh. Right on both counts, reverting.

**Pull request merged**